src/Drawing.ts

```diff
@@ -0,0 +1,654 @@
+/**
+ * Adapted from https://github.com/purescript-contrib/purescript-drawing
+ *
+ * @since 1.0.0
+ */
+import * as B from 'fp-ts/lib/boolean'
+import * as IO from 'fp-ts/lib/IO'
+import * as M from 'fp-ts/lib/Monoid'
+import * as O from 'fp-ts/lib/Option'
+import * as ROA from 'fp-ts/lib/ReadonlyArray'
+import { flow } from 'fp-ts/lib/function'
+import { pipe } from 'fp-ts/lib/pipeable'
+
+import * as C from './Canvas'
+import { toCss, Color } from './Color'
+import { showFont, Font } from './Font'
+import { Point, Shape } from './Shape'
+
+const readonlyArrayMonoidDrawing = ROA.getMonoid<Drawing>()
+const getFirstMonoidColor = O.getFirstMonoid<Color>()
+const getFirstMonoidNumber = O.getFirstMonoid<number>()
+const getFirstMonoidPoint = O.getFirstMonoid<Point>()
+
+/**
+ * Represents a shape that can be drawn to the canvas.
+ *
+ * @since 1.0.0
+ */
+export type Drawing = Fill | Outline | Text | Many | Scale | Translate | Rotate | Clipped | WithShadow
+
+/**
+ * Represents the styles applied to a filled `Shape`.
+ *
+ * @since 1.0.0
+ */
+export interface FillStyle {
+  /**
+   * The fill color.
+   */
+  readonly color: O.Option<Color>
+}
+
+/**
+ * Constructs a `FillStyle`.
+ *
+ * @since 1.0.0
+ */
+export const fillStyle: (color: Color) => FillStyle = (c) => ({ color: O.some(c) })
+
+/**
+ * Gets a `Monoid` instance for `FillStyle`.
+ *
+ * @since 1.0.0
+ */
+export const monoidFillStyle = M.getStructMonoid<FillStyle>({
+  color: getFirstMonoidColor
+})
+
+/**
+ * Represents the styles applied to an outlined `Shape`.
+ *
+ * @since 1.0.0
+ */
+export interface OutlineStyle {
+  /**
+   * The outline color.
+   */
+  readonly color: O.Option<Color>
+
+  /**
+   * The outline line width.
+   */
+  readonly lineWidth: O.Option<number>
+}
+
+/**
+ * Gets a `Monoid` instance for `OutlineStyle`.
+ *
+ * @example
+ * ```typescript
```

Comment:
you can remove the fence

```diff
 * @example
- * ```typescript
 * import * as O from 'fp-ts/lib/Option';
 * import * as C from 'graphics-ts/lib/Color';
 * import * as D from 'graphics-ts/lib/Drawing';
```

Reply:
Didn't realize these were not necessary! Thanks 👍. See https://github.com/gcanti/graphics-ts/pull/7/commits/aaf3a44f1d61e72ab05ea1e2a41064c86089f2e1